src/lib/transaction_snark_scan_state/transaction_snark_scan_state.ml

```diff
@@ -1333,9 +1333,10 @@ let all_work_pairs t
             | Merge ->
                 Or_error.error_string "init_stack was Merge"
           in
-          { Transaction_witness.first_pass_ledger = first_pass_ledger_witness
+          { Transaction_witness.Stable.Latest.first_pass_ledger =
+              first_pass_ledger_witness
           ; second_pass_ledger = second_pass_ledger_witness
-          ; transaction
+          ; transaction = Transaction.read_all_proofs_from_disk transaction
```

Comment:
It feels very sad to do this. Can we potentially push up the proof-cached transactions up into this type instead of reading here? I don't think we need them until we've chosen snark work and are about to send it to the snark worker.